accepted/010-database.md

```diff
@@ -0,0 +1,41 @@
+# SQLite for Auraed
+
+This feature will accomplish **data persistency after reboots/restarts/failure** because **auraed is a systems daemon built for humans who often make mistakes**.
+
+### Outcomes
+
+  - Allow **persistent records** to exist after a daemon crash, failure.
+  - Allow **persistent records** to exist atomicaly after a gRPC request return. 
+      - There will be a deliberate coupling between the gRPC server and the data layer/model.
+  - Allow **encryption at rest** which can be unlocked using the `/etc/aurae/pki/server.key` (or a cryptographic hash) server key for the database.
+  - Allow **joining** and **connecting** records and schemas. For example we should have a collection of `VirtualMachine` records, which need to be mapped to `NIC` records.
+  - Allow **for a single database file** which can easily be backed up, recovered, and snapshotted for disaster recovery.
+  - Alow **embedding** the database directly into the `auraed` daemon such that another server/client relationship does not need to be maintained, or such that auraed will scheduele the database directly.
+
+### Goals
+
+ - All **auraed** daemons will ship with this **database**.
+ - All **auraed** daemons will have an upgrae/migration strategy of the data **in scope** for the entire project
+ - All **aurae project** components will adhere to a version number that is indicitive of the data stored in this database and semver.
+ - It will be impossible to recover a system without a key.
+ - It will be impossible to access or mutate data without a key.
+
+### Decisions
+
+ - The project will use **SQLite** to serve as the most fundamental **storage layer**.
```

Comment:
this seems fine for now…it’s future iterations where this may change…or get more complex….

I dunno if it fits in this ADR but some thought exercises to validate sqlite is _enough_ against some reasonably probable failure scenarios would be good. Maybe that already happened and it is just not documented